packages/terra-menu/src/_MenuContent.jsx

```diff
@@ -0,0 +1,171 @@
+import List from 'terra-list';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import ContentContainer from 'terra-content-container';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import Arrange from 'terra-arrange';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import MenuUtils from './_MenuUtils';
+import styles from './Menu.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Title the should be displayed in header
+   */
+  title: PropTypes.string,
+  /**
+   * Callback function for when back button is clicked
+   */
+  onRequestBack: PropTypes.func,
+  /**
+   * Callback function for when close button is clicked
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * Callback function that takes the content to be displayed next and is called when an item with nested content is clicked
+   */
+  onRequestNext: PropTypes.func.isRequired,
+  /**
+   * Menu Items/Menu Groups/Menu Dividers to be displayed
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Index within the Menu Stack
+   */
+  index: PropTypes.number.isRequired,
+};
+
+const defaultProps = {
+  children: [],
+  title: '',
+};
+
+const childContextTypes = {
+  isSelectableMenu: PropTypes.bool,
+};
+
+class MenuContent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.wrapOnClick = this.wrapOnClick.bind(this);
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.buildHeader = this.buildHeader.bind(this);
+    this.isSelectable = this.isSelectable.bind(this);
+    this.onKeyDownBackButton = this.onKeyDownBackButton.bind(this);
+  }
+
+  getChildContext() {
+    return { isSelectableMenu: this.isSelectable() };
+  }
+
+  onKeyDownBackButton() {
+    if (event.nativeEvent.keyCode === MenuUtils.KEYCODES.ENTER || event.nativeEvent.keyCode === MenuUtils.KEYCODES.SPACE) {
+      this.props.onRequestBack();
+    }
+  }
+
+  isSelectable() {
+    for (let i = 0; i < this.props.children.length; i += 1) {
+      const child = this.props.children[i];
+      if (child.props.children || child.props.isSelectable) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  wrapOnClick(item) {
+    const onClick = item.props.onClick;
+    return (event) => {
+      this.props.onRequestNext(item);
+
+      if (onClick) {
+        onClick(event);
+      }
+    };
+  }
+
+  wrapOnKeyDown(item) {
+    const onKeyDown = item.props.onKeyDown;
+    return ((event) => {
+      if (event.nativeEvent.keyCode === MenuUtils.KEYCODES.ENTER || event.nativeEvent.keyCode === MenuUtils.KEYCODES.SPACE) {
+        this.props.onRequestNext(item);
+      }
+
+      if (onKeyDown) {
+        onKeyDown(event);
+      }
+    });
+  }
+
+  buildHeader() {
+    const closeIcon = <IconClose />;
+    const closeButton = this.props.onRequestClose ? (
+      <button className={cx(['header-button', 'close-button'])} onClick={this.props.onRequestClose}>
```

Comment:
Does this allow for keyboard navigation? I am wondering if we want onChange

Reply:
Yes keyboard navigation is built into button components. What kind of use case are you thinking that onChange would be needed?

Reply:
I was wondering if the close button allowed for keyboard changes. I just tested it and all seems to work, however, I am noticing when I navigate to a submenu from the keyboard, the sub menu shows, but does not gain focus until the remaining main menu has been tabbed through.

Reply:
Yes. That's an issue with the slide-group. It's logged here: https://github.com/cerner/terra-core/issues/647